tests/test_official_zan_safety_matrix.py

```
@@ -0,0 +1,207 @@
+
+    def geturl(self) -> str:
+        return self._final_url
+
+    def read(self, limit: int = -1) -> bytes:
+        if limit < 0:
+            return self._payload
+        return self._payload[:limit]
+
+
+def _patch_open(monkeypatch: pytest.MonkeyPatch, response: _Response) -> None:
+    def fake_open(*_args, **_kwargs):
```

Ruff (0.16.2)

[warning] Missing return type annotation for private function `fake_open`

(ANN202)